urlparse4/cgurl.pyx

```diff
@@ -333,13 +392,157 @@ def urljoin(base, url, allow_fragments=True):
     This function intends to replace urljoin from urllib,
     which uses Resolve function from class GURL of GURL chromium
     """
+    str_input = isinstance(base, str)
+    if isinstance(url, str) != str_input:
+        raise TypeError("Cannot mix str and non-str arguments")
+
     decode = not (isinstance(base, bytes) and isinstance(url, bytes))
     if allow_fragments and base:
         base, url = unicode_handling(base), unicode_handling(url)
+        """
+        this part needs to be profiled to see if creating another GURL instance
+        here takes more time than expected?
+        """
+        if not GURL(base).is_valid():
```

Comment:
In which cases can GURL think URL is invalid? Would be good to add some short comment as well

Reply:
I will definitely work on that! Thank you for notifying me! 
But basically GURL thinks the URL is  invalid when it's in this form: http:///, #, a, ...